fonttools: update to 3.19.0

amake opened the pull request:
###### Description
Update fonttools to 3.19.0. The project has moved to GitHub. It has no required dependencies, but several optional dependencies that unlock additional functionality; I have added variants for the deps currently available in MacPorts.

mojca commented on print/fonttools/Portfile:
```diff
@@ -14,15 +15,37 @@ platforms           darwin
+    depends_lib-append port:py27-pygtk
+}
+
+variant all requires woff2 sym cocoa qt png gui {Install all available optional dependencies} {}
```
Are these just runtime dependencies or are they required at the build time already?

If these are build time dependencies as well, I suspect we'll see opportunistic linking.

How heavy are these dependencies in terms of the number of other dependencies pulled in and in terms of licence conflicts (= inability to get a binary)?

It might be nice to simplify the port (that is: less variants). But if variants are justified, that's ok.

I'm tempted to commit the first part right away and leave some time for discussion about the second part.

amake replied:
They are just runtime dependencies.

Licenses and dependencies:

- py27-brotli: MIT / none
- py27-sympy: BSD / small
- py27-pyobjc: MIT / small
- py27-pyqt5: GPL-3 / very big
- py27-reportlab: BSD / medium
- py27-pygtk: LGPL-2.1+ / very big

I don't know off the top of my head the implications of the above mix of licenses, but the Qt and GTK dependencies are quite heavy.

My main intent in adding the variants was user convenience; without the variants it's hard to discover what optional features are available, and which can be enabled within the confines of MacPorts. If you think they should be left out then that's OK with me.

mojca replied:
Oh, no, I didn't want to suggest removing the variants / dependencies, just trying to figure out if simplifying is an option.

If these are just runtime dependencies, the lines should be changed to reflect that (`depends_run`). The only sad thing is that a variant means that the port needs to be recompiled each time just for the sake of adding an additional dependency which is somewhat suboptimal.

If some dependencies are often useful and small, I would make them default (you can still put them into a variant and thus enable a slimmer installation, but make the variant default). Qt and GTK are a different story of course.

amake replied:
> If these are just runtime dependencies, the lines should be changed to reflect that (depends_run).

Thanks, I've done that now.

> If some dependencies are often useful and small, I would make them default 

This is a pretty extensive package, and I'm not an expert, but I think `+woff2` by default would be good, as it's small and webfont subsetting is probably a major use case. The others I'm not so sure about. I've now set `+woff2` by default.

amake commented:
My last reply is a bit buried since the comment thread is now attached to an outdated line of code.

To be clear, I believe I have addressed all feedback so far.

mojca commented:
It wasn't burried. I was just thinking whether there's any more elegant way to fix this without the need for rebuilding the port, but I'm not sure what it could be.
